### PR TITLE
Bump error-prone version

### PR DESCRIPTION
**Pull request by narape**
# TL;DR
Bump error-prone version and fix compile errors reported by the new version.

## Type
 - [ ] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
As part of refreshing the dependencies of the project here we are bumping error-prone static compiler version. The changes come from these new checks enabled in the recent versions of error prone:
* [DoNotMockAutoValue](https://errorprone.info/bugpattern/DoNotMockAutoValue)
* [InlineMeSuggester](https://errorprone.info/bugpattern/InlineMeSuggester)
* [MissingSummary](https://errorprone.info/bugpattern/MissingSummary)

## Tracking Issue
_NA_

## Follow-up issue
_NA_

### CODE REVIEW

File: jflyte/src/main/java/org/flyte/jflyte/ExecuteLocalLoader.java [hunk 1]
```diff
@@ -104,6 +104,12 @@
     }
   }
 
+  @SuppressWarnings("JdkObsolete") // error-prone complains about SortedSetMultimap#get
+  private static boolean isDuplicated(
+      SortedSetMultimap<String, String> sourcesByItemId, String itemId) {
+    return sourcesByItemId.get(itemId).size() > 1;
+  }
+
```
Comment (ckiosidis) [inline jflyte/src/main/java/org/flyte/jflyte/ExecuteLocalLoader.java:112]: you plan to handle these separately?

File: jflyte-google-cloud/src/main/java/org/flyte/jflyte/gcp/GoogleAuthHelper.java [hunk 2]
```diff
@@ -77,14 +77,19 @@ private Token getIdTokenFromIdTokenProvider(String targetAudience) throws IOExce
...
         .expiry(expiryAt)
         .tokenType("Bearer")
         .build();
   }
 
+  @SuppressWarnings("JdkObsolete") // error-prone complains about IdToken#getExpirationTime
+  private Instant getIdTokenExpirationTime(IdToken idToken) {
+    return idToken.getExpirationTime().toInstant();
+  }
+
```
Comment (ckiosidis) [inline jflyte-google-cloud/src/main/java/org/flyte/jflyte/gcp/GoogleAuthHelper.java:92]: you plan to handle these separately?

**Review by ckiosidis — APPROVED** [review]

File: jflyte/src/main/java/org/flyte/jflyte/ExecuteLocalLoader.java (hunk 1 above)
Comment (narape) [reply jflyte/src/main/java/org/flyte/jflyte/ExecuteLocalLoader.java]: An even newer version fixes this problem

File: flytekit-api/src/main/java/org/flyte/api/v1/Primitive.java [hunk 3]
```diff
@@ -89,7 +92,8 @@ public double float_() {
    * @return string value
    */
   @Deprecated
-  public String string() {
+  @InlineMe(replacement = "this.stringValue()")
```
Comment (ckiosidis) [inline flytekit-api/src/main/java/org/flyte/api/v1/Primitive.java:95]: Does Intellij support the functionality of InlineMe?

File: flytekit-local-engine/src/main/java/org/flyte/localengine/LocalEngine.java [hunk 4]
```diff
@@ -38,6 +39,14 @@
...
   @Deprecated
+  @InlineMe(
+      replacement =
+          "LocalEngine.compileAndExecute(template, runnableTasks, emptyMap(), inputs, NoopExecutionListener.create())",
+      imports = {
+        "org.flyte.localengine.LocalEngine",
+        "org.flyte.localengine.NoopExecutionListener"
+      },
+      staticImports = "java.util.Collections.emptyMap")
   public static Map<String, Literal> compileAndExecute(
```
Comment (ckiosidis) [inline flytekit-local-engine/src/main/java/org/flyte/localengine/LocalEngine.java:50]: very complex replacement

File: jflyte/src/test/java/org/flyte/jflyte/AuthorizationHeaderInterceptorTest.java [hunk 5]
```diff
@@ -81,11 +82,10 @@ public void authHeaderDeliveredToServer() throws Exception {
     String idToken = "awesome";
     String tokenType = "Bearer";
 
+    Token token =
+        Token.builder().tokenType(tokenType).accessToken(idToken).expiry(Instant.now()).build();
```
Comment (ckiosidis) [inline jflyte/src/test/java/org/flyte/jflyte/AuthorizationHeaderInterceptorTest.java:86]: 👍

**Review by ckiosidis — APPROVED** [review]

File: flytekit-api/src/main/java/org/flyte/api/v1/Primitive.java (hunk 3 above)
Comment (narape) [reply flytekit-api/src/main/java/org/flyte/api/v1/Primitive.java:95]: No, this is error prone doing the replacement on the caller code. This is like making the client call the replacement nor-deprecated code instead of the deprecated method.
